Sources/BraintreeCoreSwift/BTGraphQLHTTP.swift

```diff
@@ -0,0 +1,244 @@
+    }
+
+    public override func put(_ path: String, parameters: NSDictionary? = nil, completion: RequestCompletion?) {
+        NSException(name: exceptionName, reason: "PUT is unsupported").raise()
+    }
+
+    public override func delete(_ path: String, completion: @escaping RequestCompletion) {
+        NSException(name: exceptionName, reason: "DELETE is unsupported").raise()
+    }
+
+    public override func delete(_ path: String, parameters: NSDictionary? = nil, completion: RequestCompletion?) {
+        NSException(name: exceptionName, reason: "DELETE is unsupported").raise()
+    }
+
+    // MARK: - Internal methods
+    
+    func httpRequest(
+        method: String,
+        parameters: NSDictionary? = [:],
+        completion: RequestCompletion?
+    ) {
+        var errorUserInfo: [String: Any] = [:]
+
+        if self.baseURL.absoluteString.isEmpty || self.baseURL.absoluteString == "" {
+            errorUserInfo["method"] = method
+            errorUserInfo["parameters"] = parameters
+            let error = Self.constructError(code: .missingBaseURL, userInfo: errorUserInfo)
+            completion?(nil, nil, error)
+            return
+        }
+        
+        let authorization: String
+        switch clientAuthorization {
+        case .authorizationFingerprint(let fingerprint):
+            authorization = fingerprint
+        case .tokenizationKey(let key):
+            authorization = key
+        default:
+            authorization = "" 
+        }
+        
+        guard let components = URLComponents(string: baseURL.absoluteString) else {
+            let error = Self.constructError(
+                code: .urlStringInvalid,
+                userInfo: [NSLocalizedDescriptionKey: "The URL absolute string is malformed or invalid."]
+            )
+            completion?(nil, nil, error)
+            return
+        }
+
+        guard let urlFromComponents = components.url else {
+            let error = Self.constructError(
+                code: .urlStringInvalid,
+                userInfo: [NSLocalizedDescriptionKey: "The URL absolute string is malformed or invalid."]
+            )
+            completion?(nil, nil, error)
+            return
+        }
+
+        let headers = [
+            "User-Agent": userAgentString(),
+            "Braintree-Version": BTCoreConstants.graphQLVersion,
+            "Authorization": "Bearer \(authorization)",
+            "Content-Type": "application/json; charset=utf-8"
+        ]
+        
+        var request: URLRequest
+    
+        do {
+            let bodyData = try JSONSerialization.data(withJSONObject: parameters ?? [:])
+            request = URLRequest(url: urlFromComponents)
+            request.httpBody = bodyData
+            request.allHTTPHeaderFields = headers
+            request.httpMethod = method
+            
+            let task: URLSessionTask = session.dataTask(with: request) { [weak self] data, response, error in
```

Comment:
```suggestion
            // Perform the actual request
            let task: URLSessionTask = session.dataTask(with: request) { [weak self] data, response, error in
```

Nitpick comment - can we preserve this comment? It always helped drill down to the actual request, since it's pretty obfuscating and hard to find.